Assignment3/Dictionary.py

```diff
@@ -0,0 +1,28 @@
+# coding=utf-8
+class Dictionary:
+    """Dummy implementation of the Dictionary class for testing purposes.
+       In the real problem the methods isWord(string) and isPrefix(string) are assumed to have
+       θ(n) running time where n is the number of characters in the string."""
+
+    def __init__(self, word_list):
+        # In this approach we are case insensitive, thus we change all the given words and prefixes to uppercase.
+        for i in range(len(word_list)):
+            word_list[i] = word_list[i].upper()
+        # Initialise dictionary's word list
+        self.__word_list = word_list
+
+        # Populate dictionary's prefix list based on the given word list
+        self.__prefix_list = []
```

Comment:
It is changed now. So for the final word (say 'dog') isPrefix returns False. And in the code later I first check isWord() and then continue searching if isPrefix() returns True. I hope that's the change you were thinkg of 👍 I might have included it as an optimization earlier but I thought isPrefix() is supposed to be defined exactly in a way that was described in an assignment question.